agent/structs/service_definition.go

```diff
@@ -251,46 +124,6 @@ func (s *ServiceDefinition) CheckTypes() (checks CheckTypes, err error) {
 	return checks, nil
 }
 
-// ServiceDefinitionConnectProxy is the connect proxy config  within a service
-// registration. Note this is duplicated in config.ServiceConnectProxy and needs
-// to be kept in sync.
-type ServiceDefinitionConnectProxy struct {
-	Command   []string               `json:",omitempty"`
-	ExecMode  string                 `json:",omitempty"`
-	Config    map[string]interface{} `json:",omitempty"`
-	Upstreams []Upstream             `json:",omitempty"`
-}
-
-func (p *ServiceDefinitionConnectProxy) MarshalJSON() ([]byte, error) {
-	type typeCopy ServiceDefinitionConnectProxy
-	copy := typeCopy(*p)
-
-	// If we have config, then we want to run it through our proxyConfigWalker
-	// which is a reflectwalk implementation that attempts to turn arbitrary
-	// interface{} values into JSON-safe equivalents (more or less). This
-	// should always work because the config input is either HCL or JSON and
-	// both are JSON compatible.
-	if copy.Config != nil {
-		configCopyRaw, err := copystructure.Copy(copy.Config)
-		if err != nil {
-			return nil, err
-		}
-		configCopy, ok := configCopyRaw.(map[string]interface{})
-		if !ok {
-			// This should never fail because we KNOW the input type,
-			// but we don't ever want to risk the panic.
-			return nil, fmt.Errorf("internal error: config copy is not right type")
-		}
-		if err := reflectwalk.Walk(configCopy, &proxyConfigWalker{}); err != nil {
-			return nil, err
-		}
-
-		copy.Config = configCopy
-	}
-
-	return json.Marshal(&copy)
-}
-
```

Comment:
(for below chunk)

Is `proxyConfigWalker` now dead too?

Reply:
I'm uncertain - I think I may have been a bit overeager in deleting this bit, restored for now pending discussion.